janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchIndex.java

```diff
@@ -332,7 +339,7 @@ public void register(String store, String key, KeyInformation information, BaseT
                 if (mappings == null || (!mappings.isDynamic() && !mappings.getProperties().containsKey(key))) {
                     //Error if it is not dynamic and have not the property 'key'
                     throw new PermanentBackendException("The external mapping for index '"+ indexStoreName + "' and type '" + store + "' do not have property '" + key + "'");
-                } else if (mappings.isDynamic()) {
+                } else if (allowMappingUpdate && mappings.isDynamic()) {
```

Comment:
Thanks for updating this. But I'm confused because it doesn't look like any test changes were necessary as part of the update. It looks like the tests as written pass regardless of whether `allowMappingUpdate` is checked (e.g. when I remove it and just check `mappings.isDynamic()` all tests in `ElasticSearchConfigTest` still pass). I'd expect a test failure if the check is removed.

Reply:
Your are right, I did not check ES mapping after the update.